docusaurus/docs/dev-docs/upgrade-tool.md

```diff
@@ -0,0 +1,164 @@
+| [`--dry-run`](#simulate-the-upgrade-without-updating-any-files-dry-run)  | Simulate the upgrade without updating any files (default: false) |
+| [`-d, --debug`](#get-detailed-debugging-information)                     | Get more logs in debug mode (default: false)                     |
+| [`-s, --silent`](#execute-the-upgrade-silently)                          | Don't log anything (default: false)                              |
+| [`-p, --project-path`](#select-a-path-for-the-strapi-application-folder) | Path to the Strapi project                                       |
+| [`-V, --version`](#get-the-current-strapi-version)                       | Output the version number                                        |
+| [`-h, --help`](#get-help)                                                | Print command line options                                       |
+
+### Simulate the upgrade without updating any files (dry run)
+
+When passing the `--dry-run` option, the codemods are executed without actually editing the files. This allows simulating the upgrade of the codebase, checking the outcomes without applying any changes:
+
+Examples:
+
+```bash
+npx @strapi/upgrade major --dry-run
+npx @strapi/upgrade minor --dry-run
+npx @strapi/upgrade patch --dry-run
+npx @strapi/upgrade codemods --dry-run
+```
+
+### Select a path for the Strapi application folder
+
+When passing the `-p` option you can specify in which folder the Strapi application is located.
+
+Example:
+
+```bash
+npx @strapi/upgrade major -p /path/to/the/Strapi/application/folder
+```
+
+### Get the current Strapi version
+
+When passing the `--version` option (or its `-V` shorthand), the current Strapi version is logged.
+
+Example:
+
+```sh
+$ npx @strapi/upgrade -V
+4.15.1
+```
+
+### Get detailed debugging information
+
+When passing the `--debug` option (or its `-d` shorthand), the upgrade tool provides more detailed logs while running:
+
+```bash
+npx @strapi/upgrade --debug
+```
+
+### Execute the upgrade silently
+
+When passing the `--silent` option (or its `-s` shorthand), the tool executes the upgrade without providing any log:
+
+```bash
+npx @strapi/upgrade --silent
+```
+
+### Get help
+
+When passing the `--help` option (or its `-h` shorthand), help information is displayed, listing the available options:
+
+Example:
+
+```bash
+$ npx @strapi/upgrade -h
+Usage: upgrade [options]
+
+Upgrade to the desired version
+
+Options:
+  --dry-run                          Simulate the upgrade without updating any files (default: false)
+  -d, --debug                        Get more logs in debug mode (default: false)
+  -s, --silent                       Don't log anything (default: false)
+  -p, --project-path <project-path>  Path to the Strapi project
+  -V, --version                      output the version number
+  -h, --help                         Print command line options
```

Comment:
Using `upgrade --help` will display the CLI commands and options 

```text
Usage: upgrade <command> [options]

Options:
  -V, --version    output the version number
  -h, --help       Print command line options

Commands:
  major [options]  Upgrade to the next available major version of Strapi
  minor [options]  Upgrade to ...
  patch [options]  Upgrade to ...
  help [command]   Print options for a specific command
```

Whereas using `upgrade <command> --help` will display help for the actual command, displaying its options

```text
Usage: upgrade major [options]

Upgrade to the next available major version of Strapi

Options:
  -p, --project-path <project-path>  Path to the Strapi project
  -n, --dry                          Simulate the upgrade without updating any files (default: false)
  -d, --debug                        Get more logs in debug mode (default: false)
  -s, --silent                       Don't log anything (default: false)
  -h, --help                         display help for command
```